### PR TITLE
feat(js): add view pager hook

### PR DESCRIPTION
**Pull request by troZee**
# Summary

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [ ] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)

### CODE REVIEW

**Comment by bataevvlad** [comment]
https://github.com/user-attachments/assets/6b49824e-ac5d-44ec-9fab-41b11ec0d79d

File: example/src/PagerHookExample.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { useMemo } from 'react';
+import { StyleSheet, View, SafeAreaView, Text } from 'react-native';
+
+import { LikeCount } from './component/LikeCount';
+import { NavigationPanel } from './component/NavigationPanel';
+import { useViewPager } from 'react-native-pager-view';
+
+export function PagerHookExample() {
+  const { AnimatedPagerView, ref, ...rest } = useViewPager(10);
```
Comment (okwasniewski) [inline example/src/PagerHookExample.tsx:9]: I think in order to make this scalable and not introduce breaking changes in the future, a better design will be to use an object:

```ts
const { AnimatedPagerView, ref, ...rest } = useViewPager({pageCount: 10});
```

This way we can easily add more parameters later

File: src/useViewPager.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+              position: onPageScrollPosition,
+            },
+          },
+        ],
+        {
+          useNativeDriver: true,
+        }
+      ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
```
Comment (okwasniewski) [inline src/useViewPager.ts:93]: Why?

File: src/useViewPager.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
+import type * as ReactNative from 'react-native';
```
Comment (troZee) [inline src/useViewPager.ts:1]: Please rename it to `usePagerView`

**Pull request merged**